dotCMS/src/main/java/com/dotcms/rest/api/v1/secret/ServiceIntegrationHelper.java

```diff
@@ -0,0 +1,428 @@
+                        incomingParamName));
+            }
+
+            final Param incomingParam = incomingParamEntry.getValue();
+            //We revise the incoming param against the definition loaded from the yml.
+            if(describedParam.isRequired() && UtilMethods.isNotSet(incomingParam.getValue())){
+               throw new IllegalArgumentException(
+               String.format("Params named `%s` is marked as required in the descriptor but does not have any value.", incomingParamName));
+            }
+        }
+    }
+
+    /**
+     * Validate the incoming param names match the params described by a serviceDescriptor yml.
+     * This is mostly useful to validate a delete param request
+     * @param incomingParamNames
+     * @param serviceDescriptor
+     * @throws DotDataException
+     */
+    private void validateIncomingParams(final Set<String> incomingParamNames, final ServiceDescriptor serviceDescriptor)
+            throws DotDataException {
+
+        //Param/Property names are case sensitive.
+        final Map<String, Param> serviceDescriptorParams = serviceDescriptor.getParams();
+        for (final String incomingParamName : incomingParamNames) {
+
+            final Param describedParam = serviceDescriptorParams.get(incomingParamName);
+            if(serviceDescriptor.isAllowExtraParameters() && null == describedParam){
+                //if the param isn't found in our description but the allow extra params flag is true we're ok
+                continue;
+            }
+            //If the flag isn't true. Then we must reject the unknown param.
+            if(null == describedParam) {
+                throw new IllegalArgumentException(String.format(
+                        "Params named `%s` can not be matched against service descriptor. ",
+                        incomingParamName));
+            }
+        }
+    }
+
+    /**
+     * This is used to manipulate a FormDataMultiPart and extract all the files it might contain
+     * Internally the file os analyzed and then processed to create new file integrations
+     * @param multipart The multi-part form
+     * @param user Logged in dude.
+     * @throws IOException
+     * @throws DotDataException
+     */
+    void createServiceIntegration(final FormDataMultiPart multipart, final User user)
+            throws IOException, DotDataException {
+        final List<File> files = new MultiPartUtils().getBinariesFromMultipart(multipart);
+        if(!UtilMethods.isSet(files)){
+            throw new DotDataException("Unable to extract any files from multi-part request.");
+        }
+        for (final File file : files) {
+            //TODO: verify file length and kit it back if exceeds a max
+            try(final InputStream inputStream = Files.newInputStream(Paths.get(file.getPath()))){
+                serviceIntegrationAPI.createServiceDescriptor(inputStream, user);
+            }catch (Exception e){
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4623512)